### ISSUE TITLE
http 415 when patching the status of an un-typed resource

### ISSUE THREAD

**Issue by lburgazzoli** [fabric8io/kubernetes-client#3379]
in my integration tests (against the k8s mock web server), I have to update the status of an unstructured object, so I'm doing something like:

```java
client.
    .customResource(ctx)
    .inNamespace(namespace)
    .withName(name)
    .updateStatus(Serialization.asJson(unstructured));
```

this was working on fabric8 client 5.4.x (with a different dsl) but now, I'm getting HTTP 415
And a message that says:

```
Caused by: io.fabric8.kubernetes.client.KubernetesClientException:
    Failure executing: PATCH 
    at: http://localhost:49253/apis/camel.apache.org/v1alpha1/namespaces/test/foo/bar/status
```

**Comment by shawkins** [comment]
To avoid a deprecated method call the RawCustomResourceOperationsImpl updateStatus method is now calling patchStatus - but that defaults to a json merge patch, which is not yet supported by the mock logic.  The RawCustomResourceOperationsImpl should probably go back to calling the deprecated updateStatus method.

**Comment by lburgazzoli** [comment]
@shawkins thx for digging into it, is there a workaround for this ?

**Comment by shawkins** [comment]
Not through RawCustomResourceOperationsImpl and the mock server, no.  The genericKubernetesResource will work and it also exposes the newer patch methods.

**Comment by lburgazzoli** [comment]
@shawkins that's great so yes, using `genericKubernetesResource` works.

**Comment by gdauchet** [comment]
> To avoid a deprecated method call the RawCustomResourceOperationsImpl updateStatus method is now calling patchStatus - but that defaults to a json merge patch, which is not yet supported by the mock logic. The RawCustomResourceOperationsImpl should probably go back to calling the deprecated updateStatus method.

I have a question in https://github.com/fabric8io/kubernetes-client/pull/3385 you commented that it would be better to go back to the deprecated method but my PR is trying to support json merge patch in the mock, is not it the good solution ?

**Comment by shawkins** [comment]
> I have a question in #3385 you commented that it would be better to go back to the deprecated method but my PR is trying to support json merge patch in the mock, is not it the good solution ?

As mentioned on the pr it's more like a separate enhancement, not a specific fix for this issue.  patchStatus is sementically different than updateStatus, so the direct fix is change the raw calls back to updateStatus.

**Comment by gdauchet** [comment]
> > I have a question in #3385 you commented that it would be better to go back to the deprecated method but my PR is trying to support json merge patch in the mock, is not it the good solution ?
> 
> As mentioned on the pr it's more like a separate enhancement, not a specific fix for this issue. patchStatus is sementically different than updateStatus, so the direct fix is change the raw calls back to updateStatus.

Oki thanks so I will create an issue about supporting json merge patch in the mock and do a corresponding issue. And it will also solve this one

**Comment by manusa** [comment]
Fixed by #3395
